addons/purchase_stock/models/stock.py

```diff
@@ -105,6 +105,20 @@ def _get_related_invoices(self):
         rslt += self.mapped('picking_id.purchase_id.invoice_ids').filtered(lambda x: x.state == 'posted')
         return rslt
 
+    def _decrease_initial_demand(self, qty, stream='UP'):
+        done_move_to_return = []
+        returned_moves = self.env['stock.move']
+        for move in self:
+            if move.created_purchase_line_id and move.state not in ('done, cancel'):
+                if move.created_purchase_line_id.state == 'draft':
+                    move.created_purchase_line_id.product_qty -= qty
+                else:
+                    done_move_to_return.append((move, move.created_purchase_line_id.order_id))
+                    in_move = move.created_purchase_line_id.move_ids.filtered(lambda move: move.state not in ('done', 'cancel'))[0]
+                    extra_move_vals = in_move._prepare_extra_move_vals(qty)
```

Comment:
I don't get why an extra move is created for the receipt with the quantity to decrease.